File: src/decoders/week-menu.ts
import type { WeekMenu } from "~/models";
import { decodeFoodAllergen } from "./food-allergen";
import { decodeFoodLabel } from "./food-label";
import { decodeDomain } from "./domain";
import { decodeMenu } from "./menu";

export const decodeWeekMenu = (menu: any): WeekMenu => ({
  containsLunch: menu.AvecRepasMidi,
  containsDinner: menu.AvecRepasSoir,

  days: menu.ListeJours.V.map(decodeMenu),
  weeks: decodeDomain(menu.DomaineDePresence.V),

  allergens: menu.ListeAllergenes.V.map(decodeFoodAllergen),
  labels: menu.Listelabels.V.map(decodeFoodLabel)
});

File: library/src/commonMain/kotlin/decoders/week-menu.kt
package decoders

import kotlinx.serialization.json.*
import models.WeekMenu

fun decodeWeekMenu (menu: JsonObject) = WeekMenu(
    containsLunch = menu["AvecRepasMidi"]!!.jsonPrimitive.boolean,
    containsDinner = menu["AvecRepasSoir"]!!.jsonPrimitive.boolean,

    days = menu["ListeJours"]!!.jsonObject["V"]!!.jsonArray.map { decodeMenu(it.jsonObject) },
    weeks = decodeDomain(menu["DomaineDePresence"]!!.jsonObject["V"]!!.jsonPrimitive.content),

    allergens = menu["ListeAllergenes"]!!.jsonObject["V"]!!.jsonArray.map { decodeFoodAllergen(it.jsonObject) },
    labels = menu["Listelabels"]!!.jsonObject["V"]!!.jsonArray.map { decodeFoodLabel(it.jsonObject) }
)